crates/rust-project-goals-cli-llm/src/updates.rs

```diff
@@ -108,17 +79,16 @@ pub async fn updates(
     Ok(())
 }
 
-async fn prepare_flagship_goals(
+async fn prepare_goals(
     repository: &Repository,
     issues: &[ExistingGithubIssue],
     filter: &Filter<'_>,
-    llm: &LargeLanguageModel,
-    quick: bool,
-    updates: &mut Updates,
-) -> anyhow::Result<()> {
+    flagship: bool,
+) -> anyhow::Result<Vec<UpdatesGoal>> {
+    let mut result = vec![];
     // First process the flagship goals, for which we capture the full text of comments.
```

Comment:
also that comment it now outdated (again, I can fix that tomorrow :p)

```suggestion
    // We process flagship and regular goals in two passes, and capture comments differently for flagship goals.
```